129_functions.py

```diff
@@ -0,0 +1,205 @@
+def my_function():
+    print("This is my function!")
+    
+my_function()
+
+# II. BUILT-IN FUNCTIONS
+# List of all 69 built-in functions
+# abs()
+# delattr()
+# hash()
+# memoryview()
+# set()
+# all(
+# dict()
+# help()
+# min()
+# setattr()
+# any()
+# dir()
+# hex()
+# next()
+# slice()
+# ascii()
+# divmod()
+# id()
+# object()
+# sorted()
+# bin()
+# enumerate()
+# input()
+# oct()
+# staticmethod()
+# bool()
+# eval()
+# int()
+# open()
+# str()
+# breakpoint()
+# exec()
+# isinstance()
+# ord()
+# sum()
+# bytearray()
+# filter()
+# issubclass()
+# pow()
+# super()
+# bytes()
+# float()
+# iter()
+# print()
+# tuple()
+# callable()
+# format()
+# len()
+# property()
+# type()
+# chr()
+# frozenset()
+# list()
+# range()
+# vars()
+# classmethod()
+# getattr()
+# locals()
+# repr()
+# zip()
+# compile()
+# globals()
+# map()
+# reversed()
+# __import__()
+# complex()
+# hasattr()
+# max()
+# round()
+
+# III. USER-DEFINED FUNCTIONS
+# Define a function using the keyword def followed by the name, parenthesis then a colon
+# The body is then defined on the next indentation
+def hello():
+    print("Hello world!")
+
+    # return statement is optional
+    return
+
+
+# Call the function directly using parenthesis and passing any required arguments
+hello()
+
+
+# IV. ANONYMOUS FUNCTIONS
+# These functions are not defined with keyword "def"
+# They do not have a body and are not directly called
+
+# Usage
+# Use lambda functions to shorten cde
+
+# Define an anonymous function with keyword lambda and 'n' number of arguments
+# lambda [arg1, arg2, ... argn]:expression
+
+# The following is an example of the same function defined using a standard function and
+# it's lambda equivalent
+
+# Normal function
+def linear_expression(x):
+    return (3 * x) + 2
+
+
+print(linear_expression(2))
+
+# Lambda function with the same functionality
+linear_expression_lambda = lambda x: (3 * x) + 2
+
+print(linear_expression_lambda(2))
+
+
+# V. CALLING A FUNCTION
+# Calling a function is executing a previously defined function
+def greet(name):
+    print("Hello, " + name + ".")
+
+
+# The next line is calling the defined function greet()
+greet("Ryan")
+
+
+# VI. FUNCTION ARGUMENTS
+# There a 4 types of function arguments
+#   1. Required arguments
+#   2. Keyword arguments
+#   3. Default arguments
+#   4. Variable-length arguments
+
+
+# 1. Required arguments
+# These arguments are passed in sequential order
+# On a function call, the number of arguments must match with the function definition
+def addition(a, b):
+    sum = a + b
+    print(f"{a} + {b} =", sum)
+
+
+addition(5, 6)
+
+
+# 2. Keyword arguments
+# Caller identifies arguments by argument name
+def language(language_name):
+    print("We are learning:", language_name)
+
+
+language(language_name="Python")
+
+
+# 3. Default arguments
+# A function called with no arguments uses a default value if it is defined with arguments
+def country(country_name = "USA"):
```

Comment:
```suggestion
def country(country_name="USA"):
```
Never put spaces around the `=` when specifying default arguments or passing keyword args during a function call (e.g. `enumerate(it, start=1)`).